src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp

```diff
@@ -38,162 +34,69 @@ NonClientIslandWindow::~NonClientIslandWindow()
-void NonClientIslandWindow::SetNonClientHeight(const int contentHeight) noexcept
-{
-    _titlebarUnscaledContentHeight = contentHeight;
-}
-
-// Method Description:
-// - Gets the size of the content area of the titlebar (the non-client area).
-//   This can be padded either by the margins from maximization (when the window
-//   is maximized) or the normal window borders.
-// Return Value:
-// - A Viewport representing the area of the window which should be the titlebar
-//   content, in window coordinates.
-Viewport NonClientIslandWindow::GetTitlebarContentArea() const noexcept
+// - called when the size of the window changes for any reason. Updates the
+//   sizes of our child Xaml Islands to match our new sizing.
+void NonClientIslandWindow::OnSize()
 {
     const auto scale = GetCurrentDpiScale();
+    const auto dpi = ::GetDpiForWindow(_window);
 
-    const auto titlebarContentHeight = _titlebarUnscaledContentHeight * scale;
-    const auto titlebarMarginRight = _titlebarMarginRight;
-
-    const auto physicalSize = GetPhysicalSize();
-    const auto clientWidth = physicalSize.cx;
-
-    auto titlebarWidth = clientWidth - (_windowMarginSides + titlebarMarginRight);
-    // Adjust for maximized margins
-    titlebarWidth -= (_maximizedMargins.cxLeftWidth + _maximizedMargins.cxRightWidth);
-
-    const auto titlebarHeight = titlebarContentHeight - (_titlebarMarginTop + _titlebarMarginBottom);
-
-
-    COORD titlebarOrigin = { static_cast<short>(_windowMarginSides),
-                             static_cast<short>(_titlebarMarginTop) };
-
-    if (_isMaximized)
-    {
-        titlebarOrigin.X = static_cast<short>(_maximizedMargins.cxLeftWidth);
-        titlebarOrigin.Y = static_cast<short>(_maximizedMargins.cyTopHeight);
-    }
-
-    return Viewport::FromDimensions(titlebarOrigin,
-                                    { static_cast<short>(titlebarWidth), static_cast<short>(titlebarHeight) });
-}
+    const auto dragY = ::GetSystemMetricsForDpi(SM_CYDRAG, dpi);
+    const auto dragX = ::GetSystemMetricsForDpi(SM_CXDRAG, dpi);
 
-// Method Description:
-// - Gets the size of the client content area of the window.
-//   This can be padded either by the margins from maximization (when the window
-//   is maximized) or the normal window borders.
-// Arguments:
-// - <none>
-// Return Value:
-// - A Viewport representing the area of the window which should be the client
-//   content, in window coordinates.
-Viewport NonClientIslandWindow::GetClientContentArea() const noexcept
-{
-    MARGINS margins = GetFrameMargins();
+    RECT buttonsRect = {};
+    ::DwmGetWindowAttribute(_window, DWMWA_CAPTION_BUTTON_BOUNDS, &buttonsRect, sizeof(buttonsRect));
+    const auto minMaxWidth = (buttonsRect.right - buttonsRect.left);
+    const auto nonClientHeight = (buttonsRect.bottom - buttonsRect.top);
+    const auto dragRegionWidth = static_cast<LONG>((NonClientDragBarWidth - XAML_MOUSEHOOVER_THICKNESS_LEFT - XAML_FOCUSRECT_THICKNESS_LEFT) * scale);

```

Comment:
Is `- XAML_MOUSEHOOVER_THICKNESS_LEFT - XAML_FOCUSRECT_THICKNESS_LEFT` what's causing the tiny black line of the right side of the xaml region in the titlebar? 

Shouldn't these dimensions be included in the size that `_app.GetNonClientAreaDragBarWidth()` returns, rather than static constants?

Reply:
No, because the Focus rect or the mouse hoover rect would be cut off. The next change is to unify the background color so that this will not be visible.

Reply:
No, otherwise the hoover rect and focus rect will be clipped.
The change that paints the correct colors will fix this problem.